launcher/Application.cpp

```diff
@@ -410,6 +411,24 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
                 " " "|" " "
                 "%{if-category}[%{category}]: %{endif}"
                 "%{message}");
+        
+        if(QFile::exists("logging.ini")) {
+            // load and set logging rules
+            qDebug() << "Loading logging rules from:" << QString("%1/logging.ini").arg(dataPath);
+            INIFile loggingRules;
+            bool rulesLoaded = loggingRules.loadFile(QString("logging.ini"));
+            if (rulesLoaded) {
+                QStringList rules;
+                qDebug() << "Setting log rules:";
+                for (auto it = loggingRules.begin(); it != loggingRules.end(); ++it) {
+                    auto rule = it.key() + "=" + it.value().toString();
+                    rules.append(rule);
+                    qDebug() << "    " << rule;
+                }
+                auto rules_str = rules.join("\n");
+                QLoggingCategory::setFilterRules(rules_str);
```

Comment:
Why don't we just use Qt's built-in support for this (using the `QT_LOGGING_CONF` env var)

Reply:
Because of the rule load order precedence. Rules through the `setFilterRules` Function can be overridden by a file specified by a `QT_LOGGING_CONF`
Which can in turn be overridden by raw rules specified in `QT_LOGGING_RULES`

Also, this way we can potentially distribute a default rules file so that not every debug line spams the logs.

Reply:
I think this file should be identical to `qtlogging.ini`, including the name and the format. I also think this `INIFile` is trash, [`QSettings`](https://doc.qt.io/qt-5/qsettings.html) has support for INI files and works way better, so you can parse the `[Rules]` section with it.

Reply:
hmm, yeah. I'll move it to use `QSettings`. I'll set up a default rules file.

Reply:
ok, now it uses QSettings (supports and uses the `[Rules]` section) and checks in order:
the datapath, the `AppDataLocation` via `QStandardPaths`, next to the binary.

I added a default `qtloggining.ini` and set it up to be installed. but that part will need to be vetted that it finds the file correctly on windows and mac.